std/algebra/emulated/sw_emulated/point.go

```diff
@@ -120,9 +122,69 @@ func (c *Curve[B, S]) Add(p, q *AffinePoint[B]) *AffinePoint[B] {
 	}
 }
 
-// Double doubles p and return it. It doesn't modify p.
+// AddUnified adds p and q and returns it. It doesn't modify p nor q.
+//
+// ✅ p can be equal to q, and either or both can be (0,0).
+// (0,0) is not on the curve but we conventionally take it as the
+// neutral/infinity point as per the EVM [EYP].
+//
+// It uses the unified formulas of Brier and Joye [BriJoy02] (Corollary 1).
+//
+// [BriJoy02]: https://link.springer.com/content/pdf/10.1007/3-540-45664-3_24.pdf
+// [EYP]: https://ethereum.github.io/yellowpaper/paper.pdf
+func (c *Curve[B, S]) AddUnified(p, q *AffinePoint[B]) *AffinePoint[B] {
```

Comment:
What do you think about renaming it to `Add` instead? We don't have public `AddIncomplete` anyway and it would be better anyway if the users would just use this safe method.

We could insead rename `add` to `addIncomplete`.

Reply:
hmm usually when we say `Add` in Weierstrass EC it means implicitly that the formulas are not unified nor complete so one would expect a public method `Double`. I was rather thinking of re-naming it to `AddComplete` because we deal with the `(0,0)` case but it's not a "natural" completeness — so keeping `AddUnified` makes sense IMO.

Reply:
I think in this case adding 
```go
func (c *Curve[B, S]) Add(p *AffinePoint[B]) *AffinePoint[B] {
    return c.AddUnified(p, p)
}

func (c *Curve[B, S]) Double(p *AffinePoint[B]) *AffinePoint[B] {
    return c.AddUnified(p, p)
}
```
Or just explaining in the package documentation that we use the unified formula which capture both point adds and doubling.  But I'm more in favour of the second option TBH.

Reply:
I think the second option is good.